examples/feature_demo/video_yuv.py

```diff
@@ -0,0 +1,267 @@
+        )
+
+
+def loop_video():
+    while True:
+        with av.open(FILENAME) as container:
+            for frame in container.decode(video=0):
+                # Reformat if necessary. If format is the same, reformat is a no-op.
+                # Otherwise, this includes a data copy and some computations.
+                # In the case of yuv420 -> rgb24, the image also takes more memory.
+                frame = frame.reformat(format=FORMAT)
+                # Cast to numpy array. This should not involve a data copy.
+                yield frame.to_ndarray()
+
+
+w, h = video_width_height()
+frame_generator = loop_video()
+
+# Create image object
+
+if FORMAT == "yuv420p":
+    # For planar yuv420, we use a 2-layer grayscale texture. The u and v planes
+    # fit in a single layer since they're half the size (in both dimensions).
+    tex = gfx.Texture(
+        size=(w, h, 2),
+        dim=2,
+        colorspace="yuv420p",
+        colorrange=COLORRANGE,
+        format="r8unorm",
+        usage=wgpu.TextureUsage.COPY_DST,
+    )
+elif FORMAT == "yuv444p":
+    tex = gfx.Texture(
+        size=(w, h, 3),
+        dim=2,
+        colorspace="yuv444p",
+        colorrange=COLORRANGE,
+        format="r8unorm",
+        usage=wgpu.TextureUsage.COPY_DST,
+    )
+else:
+    # For rgb/rgba we use an rgba texture (there is no rgb texture format).
+    tex = gfx.Texture(
+        size=(w, h, 1),
+        dim=2,
+        colorspace="srgb",
+        format="rgba8unorm",
+        usage=wgpu.TextureUsage.COPY_DST,
+    )
+
+im = gfx.Image(
+    gfx.Geometry(grid=tex),
+    gfx.ImageBasicMaterial(clim=(0, 255)),
+)
+
+# Setup the rest of the viz
+
+CanvasClass = OffscreenCanvas if OFFSCREEN else WgpuCanvas
+canvas = CanvasClass(size=(w // 2, h // 2), max_fps=999, vsync=False)
+renderer = gfx.renderers.WgpuRenderer(canvas, show_fps=True)
+scene = gfx.Scene()
+scene.add(im)
+camera = gfx.OrthographicCamera(w, h)
+camera.local.position = w // 2, h // 2, 0
+camera.local.scale_y = -1
+stats = gfx.Stats(viewport=renderer)
+
+
+def animate():
+    # Get next frame to upload
+    data = next(frame_generator)
+
+    if FORMAT == "yuv420p":
+        # Send the three planes to the texture. The y-plane goes to the first layer.
+        # The u-plane and v-plane go to the second layer, side-by side.
+        # Note that the u and v planes are just a quarter of the size of the y-plane.
+        # All planes are contiguous, so there are zero data copies.
+        y = data[:h]
+        u = data[h : h + h // 4].reshape(h // 2, w // 2)
+        v = data[h + h // 4 :].reshape(h // 2, w // 2)
+        tex.send_data((0, 0, 0), y)
+        tex.send_data((0, 0, 1), u)
+        tex.send_data((w // 2, 0, 1), v)
+    elif FORMAT == "yuv444p":
+        y = data[0]
+        u = data[1]
+        v = data[2]
+        tex.send_data((0, 0, 0), y)
+        tex.send_data((0, 0, 1), u)
+        tex.send_data((0, 0, 2), v)
```

Comment:
I think we can even just do `tex.send_data((0, 0, 0), data)` here.

Reply:
I thought this would work but it didn’t. 

Gave me garbled results.